plugins/device/cmd/nvidia/fingerprint.go

```diff
@@ -0,0 +1,244 @@
+package nvidia
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/nomad/plugins/device"
+	nvml "github.com/hashicorp/nomad/plugins/device/cmd/nvidia/nvml_client"
+)
+
+const (
+	// Attribute names for reporting Fingerprint output
+	MemoryMiBAttr          = "memory_mib"
+	PowerWAttr             = "power_w"
+	BAR1MiBAttr            = "bar1_mib"
+	DriverVersionAttr      = "driver_version"
+	CoresClockMHzAttr      = "cores_clock_mhz"
+	MemoryClockMHzAttr     = "memory_clock_mhz"
+	PCIBandwidthMBPerSAttr = "pci_bandwidth_mb/s"
+	DisplayStateAttr       = "display_state"
+	PersistenceModeAttr    = "persistence_mode"
+)
+
+// fingerprint is the long running goroutine that detects hardware
+func (d *NvidiaDevice) fingerprint(ctx context.Context, devices chan<- *device.FingerprintResponse) {
+	defer close(devices)
+
+	if d.nvmlClientInitializationError != nil {
+		d.logger.Error("Exiting fingerprinting due to problems with NVML loading")
```

Comment:
Lower case and then also display the error. I won't comment on all of them, but please follow this pattern on the other log messages.